hil-test/tests/ecc.rs

```diff
@@ -0,0 +1,984 @@
+//! ECC Test
+
+//% CHIPS: esp32c2 esp32c6 esp32h2
+
+#![no_std]
+#![no_main]
+
+use core::ops::Mul;
+
+use crypto_bigint::{
+    modular::runtime_mod::{DynResidue, DynResidueParams},
+    Encoding,
+    U192,
+    U256,
+};
+use defmt_rtt as _;
+use elliptic_curve::sec1::ToEncodedPoint;
+use esp_backtrace as _;
+#[cfg(feature = "esp32h2")]
+use esp_hal::ecc::WorkMode;
+use esp_hal::{
+    ecc::{Ecc, EllipticCurve, Error},
+    peripherals::Peripherals,
+    rng::Rng,
+    Blocking,
+};
+use hex_literal::hex;
+
+struct TestParams<'a> {
+    prime_fields: &'a [&'a [u8]],
+    nb_loop_mul: usize,
+    #[cfg(feature = "esp32c2")]
+    nb_loop_inv: usize,
+}
+
+const TEST_PARAMS_VECTOR: TestParams = TestParams {
+    prime_fields: &[
+        &hex!("fffffffffffffffffffffffffffffffeffffffffffffffff"),
+        &hex!("ffffffff00000001000000000000000000000000ffffffffffffffffffffffff"),
+    ],
+    nb_loop_mul: 10,
+    #[cfg(feature = "esp32c2")]
+    nb_loop_inv: 20,
+};
+
+struct Context<'a> {
+    ecc: Ecc<'a, Blocking>,
+    rng: Rng,
+}
+
+impl Context<'_> {
+    pub fn init() -> Self {
+        let peripherals = Peripherals::take();
+        let ecc = Ecc::new(peripherals.ECC, None);
+        let rng = Rng::new(peripherals.RNG);
+
+        Context { ecc, rng }
+    }
+}
+
+#[embedded_test::tests]
+mod tests {
```

Comment:
```suggestion
#[cfg(test)]
#[embedded_test::tests]
mod tests {
```

Reply:
Its not really useful since we "only" have tests in the crate, but I think its nice to only build tests when running `cargo test`

Reply:
I see, thanks! I changed it :)